### PR TITLE
fix: punchout improvements

### PR DESCRIPTION
**Pull request by shauke**
## PR Type

[x] Bugfix

## What Is the New Behavior?

Add route guard for 'punchout' routes depending on the feature toggle

## Does this PR Introduce a Breaking Change?

[x] No

### CODE REVIEW

File: src/app/pages/account/account-page.module.ts [hunk 1]
```diff
@@ -41,53 +47,48 @@ const accountPageRoutes: Routes = [
...
+        path: 'order-templates',
+        data: { breadcrumbData: [{ key: 'account.ordertemplates.link' }] },
+        loadChildren: () =>
+          import('../../extensions/order-templates/pages/order-templates-routing.module').then(
+            m => m.OrderTemplatesRoutingModule
+          ),
+      },
+      {
+        path: 'punchout',
+        canActivate: [FeatureToggleGuard, AuthorizationToggleGuard],
```
Comment (dhhyi) [inline src/app/pages/account/account-page.module.ts:60]: this route is now protected twice (once here, once in the routing-module (should be preferred))

**Review by dhhyi — CHANGES_REQUESTED** [review]
the second commit should have prefix "refactor"

File: src/app/pages/account/account-page.module.ts (hunk 1 above)
Comment (shauke) [reply src/app/pages/account/account-page.module.ts:60]: These are two different routes that need to be protected. This FeatureToggleGuard configuration protects the route `/account/punchout` while the configuration in `punchout-routing.module.ts` protects the route `/punchout`.

Comment (dhhyi) [reply src/app/pages/account/account-page.module.ts:60]: but the routing module is the same, right?
here its protecting `extensions/punchout/pages/punchout-routing.module` and in that routing module, there also is a feature toggle protection.
Am I misreading something? 🤔

Comment (dhhyi) [reply src/app/pages/account/account-page.module.ts:60]: Ah, now I see the reason for my confusion. There are two routing modules in one file. This violates the [rule of one](https://angular.io/guide/styleguide#rule-of-one) where we should be strict, wen it comes to Angular artifacts. (Different story with models)

The `AccountPunchoutRoutingModule` should be renamed and located to the parallel file `punchout-account-routing.module.ts` (which is an allowed pattern, as far as I can see...)

**Review by dhhyi — CHANGES_REQUESTED** [review]

File: src/app/pages/account/account-page.module.ts (hunk 1 above)
Comment (shauke) [reply src/app/pages/account/account-page.module.ts:60]: This is Angular style rule "01-01". We should definitely not mess with this first one. 😉 
Sorry for confusing you by doing such an unthinkable violation. I was thinking about separating these routes but it did not hurt me as much. I hopefully learned my lesson.
The account routing is now in it's own file.

**Review by dhhyi — APPROVED** [review]

**Pull request merged**